tests/testthat/test_extractData2.R

```diff
@@ -70,6 +70,19 @@ test_that("Extract data into factor with duplicate value labels", {
   expect_equal(out2$VAR1, out_factor2)
 })
 
+test_that("Extract data into factor with multiple value labels being NA", {
+  testM3 <- changeValLabels(testM2, varName = "VAR1", value = 2, valLabel = NA)
+  testM3$labels$valLabel[3] <- NA
+
+  suppressWarnings(out <- extractData2(testM3, labels2character = "VAR1", convertMiss = TRUE))
+  expect_equal(as.character(out$VAR1), c(1, NA_character_, NA, 2))
+
+  suppressWarnings(outF <- extractData2(testM3, labels2factor = "VAR1", convertMiss = TRUE))
+  empty_fac <- as.factor(rep(NA, 4))
+  attr(empty_fac, "label") <- "Variable 1"
+  expect_equal(outF$VAR1, empty_fac)
```

Comment:
I might miss a point here, but I have a quick question: Why should `VAR1` become an empty factor variable when it does contain two valid values in the `GADSdat`?

Reply:
That was the point I missed. 😄 Thanks for clarifying!
After I followed another reasoning error on my side: No, I wouldn't suggest a different behavior. I would, however, suggest displaying a warning. On the one hand, I find it hard to imagine a case where a user would have a value marked as labeled without it actually being labeled *and* needing it. On the other hand, if this user needs this value, I can well imagine them having a hard time figuring out that `extractData2` is the step where it gets lost.

Reply:
The issue with functions like `extractData2()` and `import_spss()` is that they are already very 'warning-heavy'. I tried `extractData2()` with `labels2factor = namesGADS()` on the BT22 data base (only the not imputed background variables) and I already get five warnings (four on ISCO variables with duplicate value labels and one on a lot of variables for which underlying values can not be preserved). If functions flood users with warnings this often has the unfortunate effect that users simply ignore all warnings anyway. This has happened frequently in practice before (users report an issue that could be explained by simply reading the warnings). Therefore - while I agree that this behavior could be noteworthy of a warning - I do not want to distract from more relevant warnings. Furthermore, in the discussed cases the function is simply doing what it is supposed to do.

Funnily, while trying out `extractData2()` on BT22 data I found some additional bugs, whose fixes I haved added to this PR. Could you review those as well?

Reply:
I agree, and I have also observed this issue with myself. I still think the user should be informed about every automatic and potentially unwanted change to the values, such as in this case. But maybe this could be addressed in a future issue that could take a more broad and integrated look at the warnings in this package.

Sure! Happy that my inquiry also had this nice little side effect. ;)